### PR TITLE
malloc nits

### PR DESCRIPTION
**Pull request by danlaine**
## Why this should be merged

Avoid memory allocations where possible

## How this works

Pretty straightforward 

## How this was tested

Existing UT

### CODE REVIEW

File: vms/avm/utxo/spender.go [hunk 1]
```diff
@@ -116,11 +116,13 @@ func (s *spender) Spend(
 	[][]*secp256k1.PrivateKey, // signers
 	error,
 ) {
-	amountsSpent := make(map[ids.ID]uint64, len(amounts))
-	time := s.clock.Unix()
+	var (
+		amountsSpent = make(map[ids.ID]uint64, len(amounts))
+		time         = s.clock.Unix()
+		ins          = []*avax.TransferableInput(nil)
```
Comment (danlaine) [inline vms/avm/utxo/spender.go:122]: `ins` and `keys` are returned so they escape to the heap

**Comment by StephenButtolph** [comment]
I know for some of these we explicitly used empty slices (rather than `nil` slices) because `json` marshalling treats them differently and we wanted `[]` in the json rather than `null`.

**Comment by danlaine** [comment]
> I know for some of these we explicitly used empty slices (rather than `nil` slices) because `json` marshalling treats them differently and we wanted `[]` in the json rather than `null`.
> 
> Admittedly we are extremely inconsistent here... But we should really just decide what we are trying to do.

Up to you what you want to do with respect to the JSON marshalling

**Comment by danlaine** [comment]
Is there anyone downstream that would be affected by us using `null` rather than `[]` in JSON?

**Comment by StephenButtolph** [comment]
> Is there anyone downstream that would be affected by us using `null` rather than `[]` in JSON?

javascript treats them differently - so changing these may be a breaking change.

File: vms/nftfx/mint_operation.go [hunk 2]
```diff
@@ -34,13 +34,13 @@ func (op *MintOperation) Cost() (uint64, error) {
 
 // Outs Returns []TransferOutput as []verify.State
 func (op *MintOperation) Outs() []verify.State {
-	outs := []verify.State{}
-	for _, out := range op.Outputs {
-		outs = append(outs, &TransferOutput{
+	outs := make([]verify.State, len(op.Outputs))
```
Comment (abi87) [inline vms/nftfx/mint_operation.go:37]: q: isn't marginally better to preallocate capacity only (`make([]verify.State,0, len(op.Outputs))`) and then append new element? I wonder if `make([]verify.State, len(op.Outputs))` waste time allocating default values

File: vms/platformvm/utxo/handler.go [hunk 3]
```diff
@@ -166,10 +166,12 @@ func (h *handler) Spend(
 	// Minimum time this transaction will be issued at
 	now := uint64(h.clk.Time().Unix())
 
-	ins := []*avax.TransferableInput{}
-	returnedOuts := []*avax.TransferableOutput{}
-	stakedOuts := []*avax.TransferableOutput{}
-	signers := [][]*secp256k1.PrivateKey{}
+	var (
+		ins          = []*avax.TransferableInput(nil)
```
Comment (abi87) [inline vms/platformvm/utxo/handler.go:170]: q: I believe most of p-chain txs do consume inputs (I assume e.g. import/export are the most common, but I may be wrong). If this is true wouldn't be more efficient to allocate ins as we currently do? (Getting an estimate for capacity would be great too)

**Review by abi87 — CHANGES_REQUESTED** [review]
just a few questions, mainly around how can we be sure that avoiding preallocation is more efficient than allocating (even with zero capacity)

**Pull request closed without merging**